server/src/ast/mod.rs

```diff
@@ -1,19 +1,86 @@
+use bigdecimal::BigDecimal;
+
+mod parser;
+
+#[derive(Copy, Clone, Debug, Default, Eq, PartialEq)]
 pub struct Localization {
-    lexeme: String,
+    //offset : usize,
     line: usize,
     column: usize,
 }
 
+impl Localization {
+    fn new() -> Self {
+        Self { line: 0, column: 0 }
+    }
+
+    pub fn at(line: usize, column: usize) -> Localization {
+        Localization { line, column }
+    }
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
 pub enum SyntaxTree {
-    Number(Number),
-    Sum(Box<(SyntaxTree, SyntaxTree)>, Vec<SyntaxTree>),
-    Product(Box<(SyntaxTree, SyntaxTree)>, Vec<SyntaxTree>),
-    Subtraction(Box<(SyntaxTree, SyntaxTree)>),
-    Division(Box<(SyntaxTree, SyntaxTree)>),
-    Negation(Box<SyntaxTree>),
+    Variable(String), // this should be a usize
```

Comment:
why?

Reply:
because i i dont care what the variable is called all that it matters is that its unique, i dont wanna waste heap space with strings (have to reserve memory for them on the heap and cloning is expensivish compared to u32)